ec_grasp_planner/src/handarm_parameters.py

```diff
@@ -220,3 +219,51 @@ def __init__(self, **kwargs):
         self['edge_grasp']['downward_force'] = 4.0
         self['edge_grasp']['sliding_speed'] = 0.04
         self['edge_grasp']['valve_pattern'] = (np.array([[0,0],[0,0],[1,0],[1,0],[1,0],[1,0]]), np.array([[0, 3.0]]*6))
+
+class RBOHandP24_pulpyWAM(RBOHandP24WAM):
+    def __init__(self, **kwargs):
+        RBOHandP24WAM.__init__(self, **kwargs)
+
+        # Define generic object parameters for surface grasp
+        self['surface_grasp']['object']['up_speed'] = 0.30
+
+        self['surface_grasp']['object']['pregrasp_transform'] = tra.concatenate_matrices(
+            tra.translation_matrix([-0.09, 0, 0.0]), tra.rotation_matrix(math.radians(20.0), [0, 1, 0]))
+
+        self['surface_grasp']['object']['post_grasp_transform'] = tra.concatenate_matrices(
+            tra.translation_matrix([0.0, 0.0, -0.05]),
+            tra.rotation_matrix(math.radians(-15.), [0, 1, 0]))
+
+        # Define generic object parameters for surface grasp
+        self['wall_grasp']['object']['lift_dist'] = 0.13  # short lift after initial contact (before slide)
+
+        # object specific parameters for apple
+        self['surface_grasp']['apple'] = self['surface_grasp']['object'].copy()
+        self['surface_grasp']['apple']['pregrasp_transform'] = tra.concatenate_matrices(
+            tra.translation_matrix([0.0, 0, 0.0]), tra.rotation_matrix(math.radians(20.0), [0, 1, 0]))
+
+        # object specific parameters for cucumber
+        self['surface_grasp']['cucumber'] = self['surface_grasp']['object'].copy()
+        self['surface_grasp']['cucumber']['pregrasp_transform'] = tra.concatenate_matrices(
+            tra.translation_matrix([-0.09, 0, 0.0]), tra.rotation_matrix(math.radians(40.0), [0, 1, 0]))
+
+        self['surface_grasp']['cucumber']['post_grasp_transform'] = tra.concatenate_matrices(
+            tra.translation_matrix([0.0, 0.0, -0.14]),
+            tra.rotation_matrix(math.radians(-70.), [0, 1, 0]))
+
+        # object specific parameters for punnet
+        self['surface_grasp']['punnet'] = self['surface_grasp']['object'].copy()
+        self['surface_grasp']['punnet']['pre_grasp_velocity'] = np.array([0.12, 0.06])
+
+        self['surface_grasp']['punnet']['pregrasp_transform'] = tra.concatenate_matrices(
+            #tra.translation_matrix([-0.09, 0, -0.0]), tra.rotation_matrix(math.radians(40.0), [0, 1, 0])) #okayish ... one success
+            #tra.translation_matrix([-0.1, -0.04, -0.0]), tra.rotation_matrix(math.radians(45.0), [0, 1, 0]))
+            tra.translation_matrix([-0.1, -0.02, -0.0]), tra.rotation_matrix(math.radians(35.0), [0, 1, 0])) # <-- best so far
+            #tra.translation_matrix([-0.09, -0.02, -0.0]), tra.rotation_matrix(math.radians(35.0), [0, 1, 0]))
+
+        self['surface_grasp']['punnet']['downward_force'] = 10  # important!
```

Comment:
why is this important?

Reply:
It helps to really fix the object and allows the hand to wrap around the punnet such that it is stable. With lower values the grasps were almost always failing because the hand wasn't spread enough. The comment is basically a reminder, in case someone is planning to tune the punnet again. I can remove or extend the comment. I don't know, what was the intention of your comment?